### PR TITLE
Add include_storecredit to customers

### PR DESCRIPTION
**Pull request by maxheyn**
https://developer.bigcommerce.com/api-reference/761ec193054b6-get-all-customers

We need to get the store credit for a customer but bigc wasn't allowing that. This adds support for that functionality.

Screen Shot 2022-08-09 at 4 53 23 PM
Screen Shot 2022-08-09 at 4 53 37 PM

### CODE REVIEW

File: bigc/resources/customers.py [hunk 1]
```diff
@@ -10,24 +10,31 @@ class BigCommerceCustomersAPI:
...
         """Return an iterator for all customers"""
         url_parts = urlparse('/customers')
 
         query_dict = {}
         if id_in is not None:
             query_dict['id:in'] = ','.join(map(str, id_in))
+        if include_storecredit:
+            query_dict['include'] = 'storecredit'
         if include_formfields:
             query_dict['include'] = 'formfields'
```
Comment (fieryprophet) [inline bigc/resources/customers.py:24]: If someone sets the request to include store credit and form fields, wouldn't this overwrite the setting for store credit?

File: bigc/resources/customers.py [hunk 2]
```diff
@@ -10,24 +10,31 @@ class BigCommerceCustomersAPI:
...
+            include_storecredit: bool = False) -> dict:
         """Get a specific customer by its ID"""
         url_parts = urlparse('/customers')
 
         query_dict = {'id:in': customer_id}
+
+        if include_storecredit:
+            query_dict['include'] = 'storecredit'
         if include_formfields:
             query_dict['include'] = 'formfields'
```
Comment (fieryprophet) [inline bigc/resources/customers.py:39]: Same as above.

**Review by fieryprophet — CHANGES_REQUESTED** [review]

File: bigc/resources/customers.py (hunk 1 above)
Comment (r-thomson) [reply bigc/resources/customers.py:24]: See https://github.com/MedShift/bigc/blob/ced475b05d9424d7612e6ea4c3c4eb056d018fe1/bigc/resources/products.py#L15-L24

**Review by r-thomson — CHANGES_REQUESTED** [review]

**Review by fieryprophet — APPROVED** [review]

File: bigc/resources/customers.py [hunk 3]
```diff
@@ -10,26 +10,42 @@ class BigCommerceCustomersAPI:
...
             query_dict['id:in'] = ','.join(map(str, id_in))
+        if include_storecredit:
+            include.append('storecredit')
         if include_formfields:
-            query_dict['include'] = 'formfields'
+            include.append('formfields')
+
+        if include:
+            query_dict['include'] = ','.join(include)
         url_parts = url_parts._replace(query=urlencode(query_dict))
```
Comment (r-thomson) [inline bigc/resources/customers.py:30]: Reorganized slightly to group related code together more closely
```py
        query_dict = {}

        if id_in is not None:
            query_dict['id:in'] = ','.join(map(str, id_in))

        include = []
        if include_storecredit:
            include.append('storecredit')
        if include_formfields:
            include.append('formfields')
        if include:
            query_dict['include'] = ','.join(include)

        url_parts = url_parts._replace(query=urlencode(query_dict))
```

**Review by r-thomson — CHANGES_REQUESTED** [review]

**Review by r-thomson — APPROVED** [review]

**Pull request merged**